### PR TITLE
[OpenMP][mlir] Add translation for `if` in `omp.teams`

### PR DESCRIPTION
**Pull request by shraiysh**
This patch adds translation for `if` clause on `teams` construct in OpenMP Dialect.

### CODE REVIEW

File: mlir/test/Target/LLVMIR/openmp-teams.mlir [hunk 1]
```diff
@@ -235,3 +235,51 @@ llvm.func @omp_teams_num_teams_and_thread_limit(%numTeamsLower: i32, %numTeamsUp
...
+llvm.func @duringTeams()
+llvm.func @afterTeams()
+
+// CHECK-LABEL: @teams_if
+// CHECK-SAME: (i1 [[ARG:.+]])
+llvm.func @teams_if(%arg : i1) {
+    // CHECK-NEXT: call void @beforeTeams()
+    llvm.call @beforeTeams() : () -> ()
+    // CHECK: [[NUM_TEAMS_UPPER:%.+]] = select i1 [[ARG]], i32 0, i32 1
+    // CHECK: [[NUM_TEAMS_LOWER:%.+]] = select i1 [[ARG]], i32 0, i32 1
```
Comment (kiranchandramohan) [inline mlir/test/Target/LLVMIR/openmp-teams.mlir:251]: Why is it `0` if `ARG` is true?

Comment (shraiysh) [reply mlir/test/Target/LLVMIR/openmp-teams.mlir:251]: Zero does not mean zero teams. The runtime call interprets zero as "not-specified". So, in that case, an "implementation-defined" number of teams are created.

When the condition is true, and there is no num_teams clause, the number of teams should be whatever they would be without the if clause - which in this case falls to the implementation defined number of teams. For LLVM OpenMP library, internally, it sets number of teams to one if zero is passed here.

Comment (kiranchandramohan) [reply mlir/test/Target/LLVMIR/openmp-teams.mlir:251]: Please add a comment in the test regarding this.

**Review by kiranchandramohan — APPROVED** [review]
LG.

**Pull request merged**